packages/core/src/tools/create-file-stasher.js

```diff
@@ -186,6 +190,24 @@ const ensureUploadMaxSizeNotExceeded = (streamOrData, length) => {
   }
 };
 
+// S3's max metadata size is 2KB
+// If the filename needs to be encoded, both the filename
+// and encoded filename are included in the Content-Disposition header
+const ensureMetadataMaxSizeNotExceeded = (filename) => {
+  const filenameMaxSize = ENCODED_FILENAME_MAX_LENGTH;
```

Comment:
nitpick: looks like we only use this variable once, as we directly use `ENCODED_FILENAME_MAX_LENGTH` for the `Error`, could we remove this variable?